### PR TITLE
Add pre-commit config and flake8 config

### PR DESCRIPTION
**Pull request by consideRatio**
I'd like to adopt a pre-commit config similar to many sections across the jupyterhub org, with prettier and black being the most relevant hooks added for autoformatting of markdown, yaml, and python files.

I'd like for this to not be merged before #44 or #46, but instead fix the merge conflicts in this PR.

```
# a command to re-commit the autoformatting stuff
git reset --hard HEAD~5; pre-commit run -a black; git commit -am "pre-commit: run black"; pre-commit run -a prettier; git commit -am "pre-commit: run prettier"; pre-commit run -a end-of-file-fixer; git commit -am "pre-commit: run end-of-file-fixer"; pre-commit run -a requirements-txt-fixer; git commit -am "pre-commit: run requirements-txt-fixer"; pre-commit run -a reorder-python-imports; git commit -am "pre-commit: run reorder-python-imports"
```

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# A flake8 linting config adjusted for black autoformatting, defined mainly
+# based on the following documentation:
+#
+# https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length
+#
+[flake8]
+max-line-length = 88
```
Comment (yuvipanda) [inline .flake8:7]: I usually steal from https://github.com/jupyterhub/jupyterhub/blob/main/.flake8, but stricter (see https://github.com/2i2c-org/pilot-hubs/blob/master/.flake8). In particular, I don't want flake8 to impose a line length limit...

**Review by yuvipanda — CHANGES_REQUESTED** [review]
One comment about flake8 style, otherwise good!

File: .flake8 (hunk 1 above)
Comment (consideRatio) [reply .flake8]: I have stolen that often as well in the past, but after working with @mathbunnyru in jupyter/docker-stacks, I learned that a lot was covered automatically by using black's autoformatting and that it therefore could be reduced to something like this.

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+; A flake8 linting config adjusted for black autoformatting, defined mainly
+; based on the following documentation:
+;
+; https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length
+;
+[flake8]
+; max-line-length = 88
```
Comment (yuvipanda) [inline .flake8:7]:
```suggestion
```

**Comment by yuvipanda** [comment]
Thanks a lot, @consideRatio

**Pull request merged**